packages/e2e/src/specs/messages/messages-thread-creation.e2e.ts

```diff
@@ -0,0 +1,331 @@
+      setTimeout(done, 10000)
+    })
+    await ModalGeneralPage.clickUpdateAvailableModalCloseButton()
+    await waitForClickableAndClick(await NavigationTabs.contactsTab)
+    await addNewContact(
+      existingContactName,
+      existingContactSurname,
+      existingContactPrimaryNumber
+    )
+    await waitForClickableAndClick(await NavigationTabs.messagesTab)
+  })
+  it("Press phone book button (contact exist)", async () => {
+    await waitForClickableAndClick(await MessagesPage.newMessageButton)
+    await waitForClickableAndClick(
+      await MessagesConversationPage.browseContactsButton
+    )
+    const contactName = await BrowseContactsModal.modalContactNameText
+    const contactPrimaryNumber =
+      await BrowseContactsModal.modalContactPrimaryNumberText
+
+    //TO BE UPDATED AFTER CONTACT NAME HAS DEDICATED DATA-TESTID ATTRIBUTE
+    //await expect(contactName).toHaveText(
+    //existingContactName + " " + existingContactSurname
+    //)
+    await expect(contactPrimaryNumber).toHaveText(existingContactPrimaryNumber)
+  })
+  it("Click available contact and check conversation screen has contact name and phone number displayed above messages", async () => {
+    await waitForClickableAndClick(
+      await BrowseContactsModal.modalContactNameText
+    )
+    const recipientName =
+      await MessagesConversationPage.conversationRecipientNameText
+    const recipientNameText = await recipientName.getText()
+    const recipientNumber =
+      await MessagesConversationPage.conversationRecipientPhoneText
+    const recipientNumberText = await recipientNumber.getText()
+
+    const threadName = await MessagesPage.threadRow
+    await expect(threadName).toHaveText(recipientNameText)
+    await expect(recipientNumberText).toEqual(existingContactPrimaryNumber)
+  })
+
+  it("Insert text to message input, send it and check conversation screen has contact name and phone number displayed above messages", async () => {
+    await MessagesConversationPage.insertTextToMessageInput(
+      "Check conversation title is the same"
+    )
+
+    await waitForClickableAndClick(
+      await MessagesConversationPage.sendMessageButton
+    )
+    const recipientName =
+      await MessagesConversationPage.conversationRecipientNameText
+    const recipientNameText = await recipientName.getText()
+    const recipientNumber =
+      await MessagesConversationPage.conversationRecipientPhoneText
+    const recipientNumberText = await recipientNumber.getText()
+
+    await expect(recipientNameText).toEqual(
+      existingContactName + " " + existingContactSurname
+    )
+    await expect(recipientNumberText).toEqual(existingContactPrimaryNumber)
+  })
+
+  it("click new message and insert part of existing contact name into search contact input", async () => {
+    await waitForClickableAndClick(await MessagesPage.newMessageButton)
+    await MessagesConversationPage.insertTextToSearchContactInput(
+      existingContactName.slice(0, -2)
```

Comment:
What does this slice to existing contact name? :)

Reply:
yeah, i know that :D But why? :D I wanted to know context, cause i think after a while there will be problem with what the heck is happening here :D I would advice to create function that does that slice and name it with a business purpose :)

Reply:
to check the partial information is enough to get matching contact suggestion